src/messaging/ExchangeMgr.h

```diff
@@ -232,8 +232,21 @@ class DLL_EXPORT ExchangeManager : public SecureSessionMgrDelegate, public Trans
     struct UnsolicitedMessageHandler
     {
         UnsolicitedMessageHandler() : ProtocolId(Protocols::NotSpecified) {}
+
+        constexpr void Reset() { Delegate = nullptr; }
+        constexpr bool IsInUse() const { return Delegate != nullptr; }
+        // Matches() only returns a sensible value if IsInUse() is true.
+        constexpr bool Matches(Protocols::Id aProtocolId, int16_t aMessageType) const
```

Comment:
should we return a predictable value is not in use (guessing false?)

so return `IsInUse() && (ProtocolId == aProtocolId) && (MessageType == aMessageType)`

Reply:
Yes, we could do that.  I checked, and the codesize is exactly the same, so compilers are doing a good job of optimizing out the redundant IsInUse() checks.

Reply:
Actually, I have no idea whether the codesize is the same: none of the code in this file is used so far, so it all gets stripped out of the final binary at the moment.....  Given that, I'd prefer to leave things as-is and then get measurements when they might mean something.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/5616 to follow up on this once this is not dead code.